src/embeddings.rs

```diff
@@ -110,6 +111,23 @@ impl PyEmbeddings {
         Ok(PyVocab::new(self.embeddings.clone()))
     }
 
+    /// Clone the Embeddings with a new embedding matrix.
+    fn clone_with_updated_matrix(&self, embedding_matrix: PyMatrix) -> PyResult<PyEmbeddings> {
+        let embeddings = self.embeddings.borrow();
+        let vocab = embeddings.vocab().to_owned();
+        let metadata = match &*embeddings {
+            EmbeddingsWrap::View(e) => e.metadata().cloned(),
+            EmbeddingsWrap::NonView(e) => e.metadata().cloned(),
+        };
+        let mut matrix = embedding_matrix.0.as_array().to_owned();
+        let norms = l2_normalize_array(matrix.view_mut());
```

Comment:
This would also compute norms for subword embeddings. Also, all word embedding should be normalized, since that's our expectation.